src/main/java/ch/njol/skript/lang/ReturnHandler.java

```diff
@@ -0,0 +1,206 @@
+		if (!(this instanceof Section))
+			throw new SkriptAPIException("loadReturnableSectionCode called on a non-section object");
+		ParserInstance parser = ParserInstance.get();
+		ReturnHandlerStack stack = parser.getData(ReturnHandlerStack.class);
+
+		SkriptEvent skriptEvent = new SectionSkriptEvent(name, (Section) this);
+		String previousName = parser.getCurrentEventName();
+		Class<? extends Event>[] previousEvents = parser.getCurrentEvents();
+		Structure previousStructure = parser.getCurrentStructure();
+		List<TriggerSection> previousSections = parser.getCurrentSections();
+		Kleenean previousDelay = parser.getHasDelayBefore();
+
+		parser.setCurrentEvent(name, events);
+		parser.setCurrentStructure(skriptEvent);
+		parser.setCurrentSections(new ArrayList<>());
+		parser.setHasDelayBefore(Kleenean.FALSE);
+		try {
+			return new ReturnableTrigger<>(
+				this,
+				parser.getCurrentScript(),
+				name,
+				skriptEvent,
+				trigger -> {
+					stack.push(trigger);
+					return ScriptLoader.loadItems(node);
+				}
+			);
+		} finally {
+			stack.pop();
+			parser.setCurrentEvent(previousName, previousEvents);
+			parser.setCurrentStructure(previousStructure);
+			parser.setCurrentSections(previousSections);
+			parser.setHasDelayBefore(previousDelay);
+		}
+	}
+
+	/**
+	 * Loads the code in the given {@link SectionNode} into a {@link ReturnableTrigger}.
+	 * <br>
+	 * This is a general method to load a section node without extra logic
+	 * done to the {@link ParserInstance}.
+	 * The calling code is expected to manage the {@code ParserInstance} accordingly, which may vary depending on
+	 * where the code being loaded is located and what state the {@code ParserInstance} is in.
+	 * @param node the section node to load
+	 * @param name the name of the trigger
+	 * @param event the {@link SkriptEvent} of the trigger
+	 * @return a returnable trigger containing the loaded section node
+	 */
+	@NonExtendable
+	default ReturnableTrigger<T> loadReturnableTrigger(SectionNode node, String name, SkriptEvent event) {
+		ParserInstance parser = ParserInstance.get();
+		ReturnHandlerStack stack = parser.getData(ReturnHandlerStack.class);
+		try {
+			return new ReturnableTrigger<T>(
+				this,
+				parser.getCurrentScript(),
+				name,
+				event,
+				trigger -> {
+					stack.push(trigger);
+					return ScriptLoader.loadItems(node);
+				}
+			);
+		} finally {
+			stack.pop();
+		}
+	}
+
+	/**
+	 * @param values the values to return
+	 */
+	void returnValues(T @Nullable [] values);
+
+	/**
+	 * @return whether this return handler may accept multiple return values
+	 */
+	boolean isSingleReturnValue();
+
+	/**
+	 * The return type of this return handler, or null if it can't
+	 * accept return values in this context (e.g. a function without a return type).
+	 * @return the return type
+	 */
+	@Nullable ClassInfo<T> returnValueType();
```

Comment:
```suggestion
	@Nullable ClassInfo<? extends T> returnValueType();
```
return value should only need to be a type of T

not sure if ReturnableTrigger needs adjusted.